templates/services/eiav2/model_oneof.mustache

```diff
@@ -0,0 +1,154 @@
+// {{classname}} - {{{description}}}{{^description}}struct for {{{classname}}}{{/description}}
+type {{classname}} struct {
+	{{#oneOf}}
+	{{#lambda.type-to-name}}{{{.}}}{{/lambda.type-to-name}} *{{{.}}}
+	{{/oneOf}}
+}
+
+{{#oneOf}}
+// {{{.}}}As{{classname}} is a convenience function that returns {{{.}}} wrapped in {{classname}}
+func {{#lambda.type-to-name}}{{{.}}}{{/lambda.type-to-name}}As{{classname}}(v *{{{.}}}) {{classname}} {
+	return {{classname}}{
+		{{#lambda.type-to-name}}{{{.}}}{{/lambda.type-to-name}}: v,
+	}
+}
+
+{{/oneOf}}
+
+// Unmarshal JSON data into one of the pointers in the struct
+func (dst *{{classname}}) UnmarshalJSON(data []byte) error {
+	var err error
+	{{#isNullable}}
+	// this object is nullable so check if the payload is null or empty string
+	if string(data) == "" || string(data) == "{}" {
+		return nil
+	}
+
+	{{/isNullable}}
+	{{#useOneOfDiscriminatorLookup}}
+	{{#discriminator}}
+	{{#mappedModels}}
+	{{#-first}}
+	// use discriminator value to speed up the lookup
+	var jsonDict map[string]interface{}
+	err = newStrictDecoder(data).Decode(&jsonDict)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON into map for the discriminator lookup")
+	}
+
+    // TEMP: hack to avoid unused imports until this PR is merged/released:
+    // https://github.com/OpenAPITools/openapi-generator/pull/20497
+    if err = validator.Validate(jsonDict); err != nil {
+        // Do nothing
+    }
```

Comment:
This `model_oneof.mustache` template was added as a workaround for a bug introduced in the generator some time after 7.4.0, which at the moment only impacts EIA v2.  The fix was merged, but missed the 7.11.0 release; when 7.12.0 is released we should be able to lose this template.